src/charm.py

```diff
@@ -54,12 +54,15 @@ def bind_address(self):
         port = self.model.config['port']
         return '0.0.0.0:{}'.format(port)
 
-    @property
-    def external_uri(self):
+    def external_uri(self, event):
         """Public URI used in http_publish_uri and http_external_uri config options"""
-        ingress = str(self.model.get_binding('graylog').network.ingress_address)
-        port = self.model.config['port']
-        return 'http://{}:{}/'.format(ingress, port)
+        try:
+            ingress = str(self.model.get_binding('graylog').network.ingress_address)
+            port = self.model.config['port']
+            return 'http://{}:{}/'.format(ingress, port)
+        except TypeError:
+            event.defer()
+            return "http://"
```

Comment:
It is not clear why you need a return if you are deferring the event.

Reply:
defer does not end execution

Reply:
> defer does not end execution

Yes I am aware. The `return "http://"` happens to be the last line of the function though.